watertap/unit_models/tests/test_unit_test_harness.py

```diff
@@ -0,0 +1,97 @@
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+import watertap.property_models.NaCl_prop_pack as props
+from watertap.unit_models.tests.unit_test_harness import UnitTestHarness
+
+from idaes.core.solvers import get_solver
+from idaes.core.util.scaling import calculate_scaling_factors
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver(options={"bound_push": 1e-8})
+
+
+# -----------------------------------------------------------------------------
+class TestReverseOsmosis0D_default(UnitTestHarness):
+    def configure(self):
+        self.model = ConcreteModel()
+        # build unit
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+        m.fs.properties = props.NaClParameterBlock()
+        m.fs.unit = ReverseOsmosis0D(
+            property_package=m.fs.properties,
+            has_pressure_change=True,
+            concentration_polarization_type=ConcentrationPolarizationType.calculated,
+            mass_transfer_coefficient=MassTransferCoefficient.calculated,
+        )
+
+        # fully specify system
+        feed_flow_mass = 1
+        feed_mass_frac_NaCl = 0.035
+        feed_mass_frac_H2O = 1 - feed_mass_frac_NaCl
+        feed_pressure = 50e5
+        feed_temperature = 273.15 + 25
+        membrane_pressure_drop = 3e5
+        length = 20
+        membrane_area = 50
+        A = 4.2e-12
+        B = 3.5e-8
+        pressure_atmospheric = 101325
+
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "NaCl"].fix(
+            feed_flow_mass * feed_mass_frac_NaCl
+        )
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "H2O"].fix(
+            feed_flow_mass * feed_mass_frac_H2O
+        )
+        m.fs.unit.inlet.pressure[0].fix(feed_pressure)
+        m.fs.unit.inlet.temperature[0].fix(feed_temperature)
+        m.fs.unit.deltaP.fix(-membrane_pressure_drop)
+        m.fs.unit.area.fix(membrane_area)
+        m.fs.unit.A_comp.fix(A)
+        m.fs.unit.B_comp.fix(B)
+        m.fs.unit.permeate.pressure[0].fix(pressure_atmospheric)
+
+        m.fs.unit.feed_side.channel_height.fix(0.002)
+        m.fs.unit.feed_side.spacer_porosity.fix(0.75)
+        m.fs.unit.length.fix(length)
+
+        # scale unit
+        m.fs.properties.set_default_scaling(
+            "flow_mass_phase_comp", 1, index=("Liq", "H2O")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mass_phase_comp", 1e2, index=("Liq", "NaCl")
+        )
+        calculate_scaling_factors(m)
+
+        self.unit_model_block = m.fs.unit
+        self.unit_statistics = {
+            "number_variables": 121,
+            "number_total_constraints": 109,
+            "number_unused_variables": 0,
+        }
+        self.unit_solutions = {
+            "m.fs.feed_side.properties_in[0].flow_mass_phase_comp[Liq, H2O]": 0.965,
+            "m.fs.feed_side.properties_in[0].flow_mass_phase_comp[Liq, NaCl]": 0.035,
```

Comment:
If we're using `eval()`, then the indices in these strings need to have their own quotes as well:

```suggestion
            "m.fs.feed_side.properties_in[0].flow_mass_phase_comp['Liq', 'H2O']": 0.965,
            "m.fs.feed_side.properties_in[0].flow_mass_phase_comp['Liq', 'NaCl']": 0.035,
```

Reply:
I would recommend instead that we just store the string location on the model, e.g.
```suggestion
            "fs.feed_side.properties_in[0].flow_mass_phase_comp[Liq, H2O]": 0.965,
            "fs.feed_side.properties_in[0].flow_mass_phase_comp[Liq, NaCl]": 0.035,
```

And then use the `find_component` method on the model object below in place of `eval` (detailed below).